runtime/dart_isolate.cc

```diff
@@ -280,8 +283,136 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRootIsolate(
+  fml::RefPtr<fml::TaskRunner> platform_task_runner =
+      task_runners.GetPlatformTaskRunner();
+  FML_DCHECK(platform_task_runner);
+
+  auto isolate_group_data =
+      std::make_unique<std::shared_ptr<DartIsolateGroupData>>(
+          std::shared_ptr<DartIsolateGroupData>(
+              *static_cast<std::shared_ptr<DartIsolateGroupData>*>(
+                  Dart_IsolateGroupData(parent_isolate))));
+
+  Settings settings((*isolate_group_data)->GetSettings());
+
+  // PlatformIsolate.spawn should behave like Isolate.spawn when unhandled
+  // exceptions happen (log the exception, but don't terminate the app). But the
+  // default unhandled_exception_callback may terminate the app, because it is
+  // only called for the root isolate (child isolates are managed by the VM and
+  // have a different error code path). So override it to simply log the error.
+  settings.unhandled_exception_callback = [errors_fatal](
+                                              const std::string& error,
+                                              const std::string& stack_trace) {
+    FML_LOG(ERROR) << "Unhandled exception:\n" << error << "\n" << stack_trace;
+    if (errors_fatal) {
+      Dart_ShutdownIsolate();
+    }
+    return true;
+  };
+
+  UIDartState::Context context(task_runners);
+  context.advisory_script_uri = (*isolate_group_data)->GetAdvisoryScriptURI();
+  context.advisory_script_entrypoint =
+      (*isolate_group_data)->GetAdvisoryScriptEntrypoint();
+  auto isolate_data = std::make_unique<std::shared_ptr<DartIsolate>>(
+      std::shared_ptr<DartIsolate>(new DartIsolate(settings,  // settings
+                                                   false,     // is_root_isolate
+                                                   true,  // is_platform_isolate
+                                                   context)));  // context
+  (*isolate_data)->platform_isolate_manager_ = platform_isolate_manager;
+
+  IsolateMaker isolate_maker =
+      [parent_isolate, debug_name](
+          std::shared_ptr<DartIsolateGroupData>* isolate_group_data,
+          std::shared_ptr<DartIsolate>* isolate_data, Dart_IsolateFlags* flags,
+          char** error) {
+        return Dart_CreateIsolateInGroup(
+            /*group_member=*/parent_isolate,
+            /*name=*/debug_name,
+            /*shutdown_callback=*/
+            reinterpret_cast<Dart_IsolateShutdownCallback>(
+                DartIsolate::SpawnIsolateShutdownCallback),
+            /*cleanup_callback=*/
+            reinterpret_cast<Dart_IsolateCleanupCallback>(
+                DartIsolateCleanupCallback),
+            /*child_isolate_data=*/isolate_data,
+            /*error=*/error);
+      };
+  Dart_Isolate platform_isolate = CreateDartIsolateGroup(
+      std::move(isolate_group_data), std::move(isolate_data), nullptr, error,
+      isolate_maker);
+
+  Dart_EnterIsolate(parent_isolate);
+
+  if (*error) {
+    return nullptr;
```

Comment:
Delete the `entry_point_handle` here (and in the other early return case below)

Or instead defer creation of the `entry_point_handle` until after `RegisterPlatformIsolate` succeeds?